cmd/fs-v1-metadata.go

```diff
@@ -237,17 +238,45 @@ func newFSMetaV1() (fsMeta fsMetaV1) {
 	return fsMeta
 }
 
-// newFSFormatV1 - initializes new formatConfigV1 with FS format info.
-func newFSFormatV1() (format *formatConfigV1) {
+// newFSFormatV2 - initializes new formatConfigV1 with FS format version 2.
+func newFSFormatV2() (format *formatConfigV1) {
 	return &formatConfigV1{
 		Version: "1",
 		Format:  "fs",
 		FS: &fsFormat{
-			Version: "1",
+			Version: "2",
 		},
 	}
 }
 
+// Supported format versions.
+var supportedFormatVersions = []string{
+	"1",
+	"2",
+	// New supported versions here.
+}
+
+// Checks if input format is version 1 and 2.
+func isFSValidFormat(formatCfg *formatConfigV1) bool {
+	if formatCfg == nil {
+		return false
+	}
+	if formatCfg.FS == nil {
+		return false
+	}
+
+	// Check for supported format versions.
+	for _, version := range supportedFormatVersions {
```

Comment:
The lexical ordering is lost if we do `<=` it is safer to not do that and do direct comparison.